internal/model/board.go

```diff
@@ -33,32 +33,107 @@ var (
 	players = []*Player{
 		{
 			Color: ColorBlue,
-			Line:  0,
-			Row:   0,
+			Position: &Coordinate{
+				Line: 0,
+				Row:  0,
+			},
 			Score: 0,
 		},
 		{
 			Color: ColorGreen,
-			Line:  -1,
-			Row:   -1,
+			Position: &Coordinate{
+				Line: -1,
+				Row:  -1,
+			},
 			Score: 0,
 		},
 		{
 			Color: ColorRed,
-			Line:  0,
-			Row:   -1,
+			Position: &Coordinate{
+				Line: 0,
+				Row:  -1,
+			},
 			Score: 0,
 		},
 		{
 			Color: ColorYellow,
-			Line:  -1,
-			Row:   0,
+			Position: &Coordinate{
+				Line: -1,
+				Row:  0,
+			},
 			Score: 0,
 		},
 	}
 	remainingPlayers = []int{0, 1, 2, 3}
 )
 
+// TileExit represents the possible exits that are available for the tile.
+type TileExit int
+
+const (
+	TileExitTop TileExit = iota
+	TileExitRight
+	TileExitBottom
+	TileExitLeft
+)
+
+func (t TileExit) Opposite() TileExit {
+	switch t {
+	case TileExitTop:
+		return TileExitBottom
+	case TileExitRight:
+		return TileExitLeft
+	case TileExitBottom:
+		return TileExitTop
+	default:
+		return TileExitRight
+	}
+}
+
+func (t TileExit) ExitCoordinate(line, row int) *Coordinate {
+	switch t {
+	case TileExitTop:
+		return &Coordinate{Line: line - 1, Row: row}
+	case TileExitRight:
+		return &Coordinate{Line: line, Row: row + 1}
+	case TileExitBottom:
+		return &Coordinate{Line: line + 1, Row: row}
+	default:
+		return &Coordinate{Line: line, Row: row - 1}
+	}
+}
+
+type TileExits []TileExit
+
+func (t TileExits) Contains(target TileExit) bool {
+	for _, exit := range t {
+		if exit == target {
+			return true
+		}
+	}
+	return false
+}
+
+// Coordinate is a coordinate on the board.
+type Coordinate struct {
+	Line int `json:"line"`
+	Row  int `json:"row"`
+}
+
+// Coordinates is set of cordinates.
+type Coordinates []*Coordinate
+
+// Contans returns whether the given Coordinate is present in the coordinate
+// array.
+func (c Coordinates) Contains(target *Coordinate) bool {
+	for _, coordinate := range c {
```

Comment:
I'm surprised go stdlib doesn't contain a way to do that

Reply:
Since Go did not support generics before 1.18, it was not implemented in stdlib sadly